pkg/cmd/synonyms/save/questions.go

```diff
@@ -0,0 +1,177 @@
+package save
+
+import (
+	"fmt"
+
+	"github.com/AlecAivazis/survey/v2"
+
+	"github.com/algolia/cli/pkg/cmd/synonyms/shared"
+	"github.com/algolia/cli/pkg/prompt"
+	"github.com/algolia/cli/pkg/utils"
+)
+
+func AskSynonym(flags *shared.SynonymFlags) error {
+	err := AskSynonymIdQuestion(flags)
+	if err != nil {
+		fmt.Println("Error when prompting synonym id survey question:", err)
+	}
+
+	err = AskSynonymTypeQuestion(flags)
+	if err != nil {
+		fmt.Println("Error when prompting synonym type survey question:", err)
+	}
+
+	switch flags.SynonymType {
+	case "", shared.Regular:
+		err := AskRegularSynonymQuestion(flags)
+		if err != nil {
+			fmt.Println("Error when prompting regular synonym survey question:", err)
+			return err
+		}
+	case shared.OneWay:
+		err := AskOneWaySynonymQuestions(flags)
+		if err != nil {
+			fmt.Println("Error when prompting one way synonym survey questions:", err)
+			return err
+		}
+	case shared.Placeholder:
+		err := AskPlaceholderSynonymQuestions(flags)
+		if err != nil {
+			fmt.Println("Error when prompting placeholder synonym survey questions:", err)
+			return err
+		}
+	case shared.AltCorrection1, shared.AltCorrection2:
+		err = AskAltCorrectionSynonymQuestions(flags)
+		if err != nil {
+			fmt.Println("Error when prompting alt correction synonym survey questions:", err)
+			return err
+		}
+	default:
+		return nil
+	}
+
+	return nil
+}
+
+func AskSynonymIdQuestion(flags *shared.SynonymFlags) error {
+	return prompt.SurveyAsk([]*survey.Question{
+		{
+			Name: "synonymId",
+			Prompt: &survey.Input{
+				Message: "id:",
+				Default: flags.SynonymID,
+			},
+			Validate: survey.ComposeValidators(survey.Required),
+		},
+	}, flags)
+}
+
+func AskSynonymTypeQuestion(flags *shared.SynonymFlags) error {
+	defaultType := flags.SynonymType
+	if flags.SynonymType == "" {
+		defaultType = shared.Regular
+	}
+
+	return prompt.SurveyAsk([]*survey.Question{{
+		Name: "SynonymType",
+		Prompt: &survey.Select{
+			Message: "type:",
+			Options: []string{shared.Regular, shared.OneWay, shared.Placeholder, shared.AltCorrection1, shared.AltCorrection2},
+			Default: defaultType,
+		},
+		Validate: survey.ComposeValidators(survey.Required),
+	}}, flags)
+}
+
+func AskRegularSynonymQuestion(flags *shared.SynonymFlags) error {
+	var synonyms string
```

Comment:
I can create a reusable function here to generate a question for comma separated values but I'd need to set a value with a  variable "key" and I'm not sure it's common in go (error prone)